### PR TITLE
Change while stretch goal, to use while loops more conventionally

### PR DESCRIPTION
**Pull request by eschwartz**
Stretch goal #7 asks the student to rewrite a `for` loop as a `while` loop. eg:

```js
let i = 0;
while(i < supplyChanges.length) {
  let item = supplyChanges[i];
  // do some conditionals, and logs....
  i++
}
```

It's an interesting exercise, but it's not a conventional use of `while` loops. At this stage, I'd rather students practice using `while` loops where they're really useful, and get a feel for how and when to use them.

----

Here's a solution to the updated the question:

```js
let partsRemaining = 572;
let boxesFilled = 0;

while (partsRemaining >= 7) {
    boxesFilled++;
    partsRemaining -= 7;
}

console.log('number of boxes filled', boxesFilled);
console.log('leftover parts', partsRemaining);
```

----

Corresponding rubric PR
https://github.com/PrimeAcademy/prime-curriculum-syllabus/pull/703

### CODE REVIEW

**Comment by DoctorHowser** [comment]
Aug 5 -- Need to discuss with @kdszafranski on or before next cohort ~3 weeks from now

**Comment by kdszafranski** [comment]
@eschwartz to update to the "boxes" idea

and the rubric.

**Comment by eschwartz** [comment]
Something like:

we have 572 parts. 
We put 7 parts in each box. 
Use a while loops to figure out how many will be left over?


**note** don't forget to update rubric

**Comment by eschwartz** [comment]
Updated the `while` stretch to match what we talked about in our team meeting. Here's a possible solution:

```js
let partsRemaining = 572;
let boxesFilled = 0;

while (partsRemaining >= 7) {
    boxesFilled++;
    partsRemaining -= 7;
}

console.log('number of boxes filled', boxesFilled);
console.log('leftover parts', partsRemaining);
```

It's a little tricky, but I think it's a nice stretch goal, and is a better way to introduce them to `while` loops

File: assignment/scripts/part-supply.js [hunk 1]
```diff
@@ -32,10 +32,14 @@ console.log('---  Stretch Goals  ---');
 // 7. Rewrite the `for` loop from #6 as a `for of` loop. 
 console.log('7. Showing supplyChanges with "for of" loop');
 
-// 8. Rewrite the `for` loop from #6 as a `while` loop.
-console.log('8. Showing supplyChanges with "while" loop');
-
-
-// 9. Write a loop to determine the total number of parts available by
+// 8. Write a loop to determine the total number of parts available by
```
Comment (eschwartz) [inline assignment/scripts/part-supply.js:35]: Swapped 8 with 9. This _total parts_ question is still dealing with the `supplyChanges` array. The boxes question is not.

**Review by kdszafranski — APPROVED** [review]
I dig it!

**Comment by eschwartz** [comment]
Thanks @kdszafranski . Can you take a look at the rubric too, so we're in sync?

https://github.com/PrimeAcademy/prime-curriculum-syllabus/pull/703

**Pull request merged**